chore: Switch from edx-sphinx-theme to sphinx-book-theme

xitij2000 opened the pull request:
The edx-sphinx theme is being deprecated, and replaced with sphinx-book-theme. This removes references to the deprecated theme and replaces them with the new standard theme for the platform.

**Testing instructions:** 
- Run `make docs` and see that the docs are generated properly and use the sphinx-book-theme

openedx-webhooks commented:
Thanks for the pull request, @xitij2000!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

mphilbrick211 commented:
Hi @openedx/arch-bom! Would someone be able to review/merge this? Thanks!

xitij2000 commented:
> Please remove unrelated requirements/ changes

Sure, I've limited it to changes to the documentation-related requirements.

xitij2000 commented:
> > > Please remove unrelated requirements/ changes
> > 
> > 
> > Sure, I've limited it to changes to the documentation-related requirements.
> 
> How did you do that? Asking because I see changes to newrelic and cryptography, which makes me a little suspicious that other changes snuck in there

Sorry, I think I misunderstood. I've limited changes to the doc.txt file which is the requirements file used only for generating documentation. It is only used and installed for generating documentation.

rgraber commented:
My bad, I was unclear. What I meant was to only include the changes that are a direct result of adding the new library. This is admittedly kind of a pain to do. 
Usually I go into the Makefile and change the `make upgrade` command by replacing `--upgrade` with `--upgrade-package pydata-sphinx-theme` and then run `make upgrade` and revert the Makefile.
I do think it's worth it, however, since in the past we've been bitten by upgrades happening in unexpected places

xitij2000 commented:
> My bad, I was unclear. What I meant was to only include the changes that are a direct result of adding the new library. This is admittedly kind of a pain to do. Usually I go into the Makefile and change the `make upgrade` command by replacing `--upgrade` with `--upgrade-package pydata-sphinx-theme` and then run `make upgrade` and revert the Makefile. I do think it's worth it, however, since in the past we've been bitten by upgrades happening in unexpected places

Sure, I've updated the PR to make the change you suggested. Unfortunately the tests are breaking after this. I'm investigating.

xitij2000 commented:
> > My bad, I was unclear. What I meant was to only include the changes that are a direct result of adding the new library. This is admittedly kind of a pain to do. Usually I go into the Makefile and change the `make upgrade` command by replacing `--upgrade` with `--upgrade-package pydata-sphinx-theme` and then run `make upgrade` and revert the Makefile. I do think it's worth it, however, since in the past we've been bitten by upgrades happening in unexpected places
> 
> Sure, I've updated the PR to make the change you suggested. Unfortunately the tests are breaking after this. I'm investigating.

Fixed it. I focussed on just running the command for docs, but later commands remove django and drf as dependencies which means for future runs those are stripped from dependent requirements as well.

rgraber reviewed (approved):
Looks great. Thanks for all your work on this.

mphilbrick211 commented:
@rgraber - is this OK to merge, or do Jeremy and Usama still need to review? If it's good-to-go, are you able to merge? Thanks!

openedx-webhooks commented:
@xitij2000 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/django-config-models&entry.867055334=https://github.com/openedx/django-config-models/pull/266&entry.1484655318=https://github.com/xitij2000&entry.752974735=2023-03-24+07:30&entry.1917517419=2023-03-29+16:19&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
